x-pack/plugins/observability_solution/investigate_app/server/models/investigation.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+
+export const investigationSchema = t.type({
```

Comment:
how about saving `alertId` as we will start investigation from alert details page?

Reply:
Good idea - So we can create an investigation from an alert, and from scratch. So you're right we need a way to store the alertId we came from. What do you think of storing an `origin` object with alertId, ruleId. Do you think we might have other way of creating an investigation from?

Reply:
I believe there will be multiple entry points to start investigation - from scratch, from dashboard/visualization, from alert, unknown/TBD. In case of dashboard/visualization, I don't know how it would work - if user can go back to same dashboard/visualization and go to ongoing investigation.

I like the idea of saving `origin` object. It could have type to indicate origin and then Id and other params. We can refine it later.

```
origin: {
    type: 'alert',
    alertId: ''
}
```
or
```
origin: {
    type: 'alert',
    alert: {
        alertId: ''
    }
}
```